### PR TITLE
feat(behavior_velocity_stop_line): replace autoware_motion_utils to autoware_trajectory

### PR DESCRIPTION
**Pull request by yhisaki**
## Description

Previously, paths represented by discrete points were manipulated using `autoware_motion_utils`. However, we plan to replace this with a class that handles paths continuously in the future by using [autoware_trajectory](https://github.com/autowarefoundation/autoware.universe/tree/main/common/autoware_trajectory).
In this PR, we implemented this replacement for behavior_velocity_stop_line.

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/9296

## How was this PR tested?

- Evaluator
  - https://evaluation.tier4.jp/evaluation/reports/9ce08fa5-0662-58c7-8541-276b4eeeb921?project_id=prd_jt
- psim

https://github.com/user-attachments/assets/857cb79e-6687-4ad8-8af8-788f9a7797f0

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp [hunk 1]
```diff
@@ -14,95 +14,87 @@
...
   // If no collision found, do nothing
-  if (!stop_point) {
+  if (!trajectory_stop_line_intersection) {
     RCLCPP_DEBUG_THROTTLE(logger_, *clock_, 5000 /* ms */, "is no collision");
     return true;
   }
 
-  const auto stop_point_idx = stop_point->first;
-  auto stop_pose = stop_point->second;
+  double stop_point_s =
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp:67]: const

File: planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/manager.cpp [hunk 2]
```diff
@@ -60,7 +61,7 @@ void TrafficLightModuleManager::modifyPathVelocity(tier4_planning_msgs::msg::Pat
   stop_reason_array.header.frame_id = "map";
   stop_reason_array.header.stamp = path->header.stamp;
 
-  first_stop_path_point_index_ = static_cast<int>(path->points.size() - 1);
+  first_stop_path_point_distance_ = autoware::motion_utils::calcArcLength(path->points);
   nearest_ref_stop_path_point_index_ = static_cast<int>(path->points.size() - 1);
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/manager.cpp:65]: Several `index` still exist. Is it hard to remove them?

Comment (yhisaki) [reply planning/behavior_velocity_planner/autoware_behavior_velocity_traffic_light_module/src/manager.cpp:65]: This PR is only for the removal of `first_stop_path_point_index_'. So this PR does not include the use of autoware_trajectory. I will replace it step-by-step.

File: planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp [hunk 3]
```diff
@@ -14,96 +14,92 @@
...
   const LineString2d stop_line = planning_utils::extendLine(
     stop_line_[0], stop_line_[1], planner_data_->stop_line_extend_length);
 
-  time_keeper_->start_track("createTargetPoint");
   // Calculate stop pose and insert index
-  const auto stop_point = arc_lane_utils::createTargetPoint(
-    *path, stop_line, planner_param_.stop_margin,
-    planner_data_->vehicle_info_.max_longitudinal_offset_m);
-  time_keeper_->end_track("createTargetPoint");
+  auto trajectory_stop_line_intersection = trajectory->crossed(stop_line.front(), stop_line.back());
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp:59]: const

File: planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp [hunk 4]
```diff
@@ -125,34 +121,22 @@ bool StopLineModule::modifyPathVelocity(PathWithLaneId * path, StopReason * stop
...
-      ego_pos_on_path.pose = stopped_pose.value();
-      ego_pos_on_path.index = findEgoSegmentIndex(path->points);
-
       // Insert stop pose
-      planning_utils::insertStopPoint(ego_pos_on_path.pose.position, ego_pos_on_path.index, *path);
-
-      debug_data_.stop_pose = stop_pose;
+      trajectory->longitudinal_velocity_mps.range(ego_on_trajectory_s, trajectory->length())
+        .set(0.0);
+      auto ego_pos_on_path = trajectory->compute(ego_on_trajectory_s).point.pose;
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp:127]: const

**Review by takayuki5168 — APPROVED** [review]

File: planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp [hunk 5]
```diff
@@ -14,96 +14,92 @@
...
   debug_data_.base_link2front = base_link2front;
-  first_stop_path_point_distance_ = autoware::motion_utils::calcArcLength(path->points);
+  first_stop_path_point_distance_ = trajectory->length();
   *stop_reason = planning_utils::initializeStopReason(StopReason::STOP_LINE);
 
   const LineString2d stop_line = planning_utils::extendLine(
     stop_line_[0], stop_line_[1], planner_data_->stop_line_extend_length);
 
-  time_keeper_->start_track("createTargetPoint");
   // Calculate stop pose and insert index
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp:58]: The comment `index` is wrong

File: planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp [hunk 6]
```diff
@@ -14,96 +14,92 @@
...
+  const double signed_arc_dist_to_stop_point = stop_point_s - ego_on_trajectory_s;
+
   switch (state_) {
     case State::APPROACH: {
       // Insert stop pose
-      planning_utils::insertStopPoint(stop_pose.position, stop_line_seg_idx, *path);
+      trajectory->longitudinal_velocity_mps.range(stop_point_s, trajectory->length()).set(0.0);
 
       // Update first stop index
-      first_stop_path_point_distance_ = autoware::motion_utils::calcSignedArcLength(
```
Comment (takayuki5168) [inline planning/behavior_velocity_planner/autoware_behavior_velocity_stop_line_module/src/scene.cpp:94]: The comment `index` is wrong.

**Review by soblin — APPROVED** [review]

**Review by rej55 — APPROVED** [review]

**Pull request merged**